compiler/resolution/preFold.cpp

```diff
@@ -456,7 +456,8 @@ static Expr* preFoldPrimOp(CallExpr* call) {
   } else if (call->isPrimitive(PRIM_IS_CLASS_TYPE)) {
     Type* t = call->get(1)->typeInfo();
 
-    if (isClassLike(t) && !t->symbol->hasFlag(FLAG_EXTERN)) {
+    if (isClassLike(t) && !t->symbol->hasFlag(FLAG_EXTERN) &&
+        !t->symbol->hasFlag(FLAG_C_PTR_CLASS)) {
```

Comment:
Should it also rule out _ddata (FLAG_DATA_CLASS)?

Reply:
I'm gonna leave that to a separate PR just to keep this one simple / focused and leverage the testing I've already done.  But I'll check it out.